src/app/directives/datatable/datatable.directive.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  AfterContentInit,
+  Directive,
+  ElementRef,
+  Input,
+  OnInit
+} from "@angular/core";
+import { DatatableComponent } from "@swimlane/ngx-datatable";
+
+@Directive({
+  selector: "[appDatatable]"
```

Comment:
suggested rename: `baw-datatable-defaults`

Reply:
Angular convention for directives is to prepend app so that it doesn't conflict with a library. My counter offer would be `appBawDatatableDefaults`

Reply:
As per https://angular.io/guide/styleguide#directive-custom-prefix

Should be: `bawDatatableDefaults`

Reply:
I suspect that because angular ships with the following lint:
```
"directive-selector": [true, "attribute", "app", "camelCase"],
"component-selector": [true, "element", "app", "kebab-case"],
```
they want you to preface the directive and component with "app".
I've updated the rule to the following:

```
"directive-selector": [true, "attribute", ["app", "baw"], "camelCase"],
```